### PR TITLE
Fix Android double-tap not working.

### LINKED ISSUE #8151: Directories cannot be entered by selecting items from a FileDialog on Android
**Operating system or device - Godot version:**
OS: Android 7.1.1 / LineageOS 14.1
Godot: Godot 2.1.2

**Issue description:**

There is no way for me to choose a folder from the directory listing and make it the current directory. If I want to change the current directory I need to enter the directory name in the top bar.

**Steps to reproduce:**
1) Make a new project with just one scene that is a FileDialog, set to filesystem access and MODE_OPEN_FILE.
2) Go to the export menu for Android and make sure to enable the "Read External Storage" permission.
3) Deploy the project to an Android device.
4) Make sure the app has been given its permissions.
5) Open the app.
6) If the currently displayed directory is empty or only contains the parent directory link, use the top bar to navigate to "/sdcard" for on-device storage.
7) Try to change the current directory to one of the listed directories by choosing it from the menu. I have tried tapping, double tapping, and hitting "Open" after tapping a directory. None of these have changed the current directory.

**Link to minimal example project:**

[filetest.zip](https://github.com/godotengine/godot/files/870390/filetest.zip) (ZIP-exported Godot project)
[filetest-apk.zip](https://github.com/godotengine/godot/files/870391/filetest-apk.zip) (Contains an exported APK)

### PR DESCRIPTION
**Pull request by madmiraal**
#37158 added the event's button state to Android's `onDoubleTap` gesture processing. However, the button state is always zero. This broke double tapping (#8151, #46100, #46101).

As per the [documentation](https://developer.android.com/reference/android/view/GestureDetector.SimpleOnGestureListener#onDoubleTap(android.view.MotionEvent)), the `onDoubleTap` notification is _"Triggered on the down event of second tap."_ Furthermore, since double-clicks are only detected on the left mouse button, even if the button state was correct, we can safely assume that the event `pressed` parameter will be `true`, `button_index` will be `MouseButton::LEFT` and `button_mask` will be `MASK_LEFT`.

Therefore this PR reverts #37158's addition of the button state to the Android `onDoubleTap` gesture processing.

Fixes #8151.
Provides a partial fix for #46100 and #46101 in as much as the output on Android will be
```
Event doubleclick = False, pressed = True, button_index = 1
Event doubleclick = False, pressed = False, button_index = 1
Event doubleclick = True, pressed = True, button_index = 1
Event doubleclick = False, pressed = True, button_index = 1
Event doubleclick = False, pressed = False, button_index = 1
```
vs the desktop output of:
```
Event doubleclick = False, pressed = True, button_index = 1
Event doubleclick = False, pressed = False, button_index = 1
Event doubleclick = True, pressed = True, button_index = 1
Event doubleclick = False, pressed = False, button_index = 1
```
In other words there is still an extra event, but at least there is a correct `doubleclick` event that can be detected appropriately.

### CODE REVIEW

File: platform/android/android_input_handler.cpp [hunk 1]
```diff
@@ -330,16 +330,15 @@ void AndroidInputHandler::_wheel_button_click(MouseButton event_buttons_mask, co
...
 	Ref<InputEventMouseButton> ev;
 	ev.instantiate();
 	_set_key_modifier_state(ev);
 	ev->set_position(p_pos);
 	ev->set_global_position(p_pos);
-	ev->set_pressed(event_button_mask != MouseButton::NONE);
-	ev->set_button_index(_button_index_from_mask(event_button_mask));
-	ev->set_button_mask(event_button_mask);
+	ev->set_pressed(true);
+	ev->set_button_index(MouseButton::LEFT);
```
[the remark this answers is not in the document]
Comment (madmiraal) [reply platform/android/android_input_handler.cpp:340]: I'm not sure supporting a right-button double-click that requires a _"non stock rom"_ to even be triggered is needed. Looking at the comments from #37158 (e.g. https://github.com/godotengine/godot/pull/37158#discussion_r396280844 and https://github.com/godotengine/godot/pull/37158#issuecomment-620063431) it appears as if you agree. Is a right-button double click needed for anything?

That been said, to get this to work properly (and properly fix #46100 and #46101) would require separating out the mouse double-clicks from the double-taps. I'm working on a PR to achieve that. In the meantime this PR provides a fix for the regression introduced by #37158.

**Comment by Calinou** [comment]
Does this PR supersede https://github.com/godotengine/godot/pull/54225?

**Comment by madmiraal** [comment]
> Does this PR supersede https://github.com/godotengine/godot/pull/54225?

Yes.

File: platform/android/java/lib/src/org/godotengine/godot/GodotLib.java [hunk 2]
```diff
@@ -106,7 +106,7 @@ public class GodotLib {
 	/**
 	 * Forward double_tap events from the main thread to the GL thread.
 	 */
-	public static native void doubleTap(int buttonMask, int x, int y);
+	public static native void doubletap(int x, int y);
```
Comment (akien-mga) [inline platform/android/java/lib/src/org/godotengine/godot/GodotLib.java:109]: Shouldn't this stay `doubleTap` in camelCase to match `double_tap` in snake case?

Comment (madmiraal) [reply platform/android/java/lib/src/org/godotengine/godot/GodotLib.java]: I've changed it back to `doubleTap` again.

**Review by m4gr3d — APPROVED** [review]

**Comment by m4gr3d** [comment]
@madmiraal Looks good; just need to be rebased prior to merging.

**Review by m4gr3d — CHANGES_REQUESTED** [review]
@madmiraal Nevermind the previous review.

I've tested the current logic on a stock Android 12 on a Pixel 5, and despite what the documentation states, double clicks from a connected bluetooth mouse is also handled by the GestureHandler.

It looks like the proper solution would be #54225  instead.

**Comment by madmiraal** [comment]
> I've tested the current logic on a stock Android 12 on a Pixel 5, and despite what the documentation states, double clicks from a connected bluetooth mouse is also handled by the GestureHandler.

This PR doesn't prevent double-clicks from working. It just assumes all double-clicks are left mouse button clicks.

**Comment by madmiraal** [comment]
I think it's a bit presumptuous to close this issue before #65434 has been approved and merged.
